### ISSUE TITLE
Colormessage patch?

### ISSUE THREAD

**Issue by Barbarossa93** [bakkeby/slock-flexipatch#3]
I was wondering if you'd be willing to replace the message patch with the [colormessage](https://tools.suckless.org/slock/patches/colormessage/) patch? Since I switch between colorschemes through Xresources (light theme during the day, dark theme at night), I'd like to be able to script a way to change the text color. Alternatively, if you are able to find a way to include text_color in the Xresources patch, that would be fantastic.

**Comment by bakkeby** [comment]
^ the above seems to work for me

**Comment by bakkeby** [comment]
You can refer to the above commit for how to use the text_color with the xresources patch.

Note that you don't have to stick with the "color4", "color6" xresources values as per the default configuration, you can change these to whatever you want. Think of it more as examples.

E.g. in my config I have:
```c
/*
 * Xresources preferences to load at startup
 */
ResourcePref resources[] = {
		{ "locked",       STRING,  &colorname[INIT] },
		{ "input",        STRING,  &colorname[INPUT] },
		{ "failed",       STRING,  &colorname[FAILED] },
		#if CAPSCOLOR_PATCH
		{ "capslock",     STRING,  &colorname[CAPS] },
		#endif // CAPSCOLOR_PATCH
		#if MESSAGE_PATCH || COLOR_MESSAGE_PATCH
		{ "message",      STRING,  &text_color },
		#endif // MESSAGE_PATCH | COLOR_MESSAGE_PATCH
};
```

and in .Xresources:
```java
! === slock ===
slock.locked:   black        /* Locked / Init */
slock.input:    bg_norm_col  /* Input */
slock.failed:   bg_sel_col   /* Failed */
slock.capslock: fg_warn_col  /* Caps Lock */
slock.message:  fg_hid_col   /* Message Text */
```

whereas the "color6" is intended to re-use colors from the terminal.

**Comment by Barbarossa93** [comment]
Oh nice, I didn't know it was that simple! Thanks @bakkeby !!

As a side question, are you familiar at all with PAM? I've enabled the pam patch but unfortunately it seems to lack the `pam_setcred` function that I need in order to use [pam-gnupg](https://github.com/cruegge/pam-gnupg). Judging from the physlock [PR](https://github.com/muennich/physlock/pull/83/commits/84260206a511c5b5aa6deada55611796a466c750) and the xsecurelock [PR](https://github.com/google/xsecurelock/commit/4f438fd63cbe8fe9333e6c0f88c0b1d089714111), it should be a simple addition. so in `slock.c`, I tried adding `pam_setcred(pamf, PAM_REFRESH_CRED)` around line 262, but it doesn't seem to work. I also set the pam service in config.h to one that should work (has been tested with i3lock and xsecurelock). Do you have any ideas?

**Comment by Barbarossa93** [comment]
One other side question, are you aware of a way to show the message immediately upon starting slock? Normally I have to start inputting text before it appears. Right now I'm using a "jiggle" script that artificially hits a key and then backspace to remove it. While this works, I have to disable failonclear which was a useful feature and so I'd rather not have to disable it

**Comment by bakkeby** [comment]
I don't really know about PAM, not sure how I would even test it.

> are you aware of a way to show the message immediately upon starting slock? 

For me it shows immediately, perhaps you have a patch enabled that conflicts with it?

**Comment by Barbarossa93** [comment]
Strange, I just went through and individually disabled each patch and I still have the same problem with the message not showing up right away. I'm clueless =/

**Comment by bakkeby** [comment]
Is your font color the same as the background when it is locked?

**Comment by Barbarossa93** [comment]
The [INPUT] and [INIT] colors are the same, as well as [PAM]. &text_color is a different color from them. To be clear, the message shows up on the INIT screen for you?

**Comment by bakkeby** [comment]
Yes it shows for me.

Btw when you do `slock -f`, does the default font of "6x10" list for you?

For me it doesn't, but I have "6x13" which I have set in my config.

**Comment by Barbarossa93** [comment]
6x10 doesnt't show and 6x13 is available to me. I tried setting the default font to be 6x13 and I still get the same problem. Otherwise, I'm using a specific font, `-spectrum-berry-medium-r-normal-sans-11-80-75-75-m-50-iso10646-1` which also shows up in `slock -f`. I guess I'll try a totally clean build of slock and go from there

**Comment by bakkeby** [comment]
Yes I don't know, it's weird that it doesn't display on the init, but only after you have started typing.

**Comment by Barbarossa93** [comment]
On a almost totally clean build (only message patch enabled) and font set to 6x13, I get the same problem. I'm wondering if maybe the XINERAMA library version is different between us? I'm running Arch linux

**Comment by Barbarossa93** [comment]
Ah, this is interesting: On a fresh reboot, the message shows up immediately. Running the locker again, the message only shows up on input

**Comment by bakkeby** [comment]
Hmm, could it have something to do with xresources?

**Comment by Barbarossa93** [comment]
I dont think so, besides color settings, everything is specific to a program. Below is the output of `xrdb -query`:
```
Xft.dpi:	96
Xft.antialias:	1
Xft.autohint:	0
Xft.hinting:	1
Xft.hintstyle:	hintfull
Xft.rgba:	rgb
Xft.lcdfilter:	lcddefault
URxvt.font:	-spectrum-berry-medium-r-normal-sans-11-80-75-75-m-50-iso10646-1, xft:tewihm:size=9:antialias=false, xft:Cozette:size=9:antialias=false, xft:SijiPlus:size=9:antialias=false, xft:Kochi Gothic:size=9:antialias=true, xft:Hack Nerd Font Mono:size=7.5:antialias=true:minspace=false, xft:Twemoji:size=7:antialias=true
URxvt.boldFont:	-spectrum-berry-bold-r-normal--11-80-75-75-m-50-iso10646-1, xft:tewihm:size=9:antialias=false, xft:Cozette:size=9:antialias=false, xft:SijiPlus:size=9:antialias=false, xft:Kochi Gothic:size=9:antialias=true, xft:Hack Nerd Font Mono:size=7.5:antialias=true:minspace=false, xft:Twemoji:size=7:antialias=true
URxvt.italicFont:	
URxvt.boldItalicFont:	
*foreground:	#5c6a72
*background:	#f8f0dc
*cursorColor:	#5c6a72
*color0:	#efead4
*color8:	#999f93
*color1:	#f85552
*color9:	#f85552
*color2:	#8da101
*color10:	#8da101
*color3:	#dfa000
*color11:	#dfa000
*color4:	#3a94c5
*color12:	#3a94c5
*color5:	#df69ba
*color13:	#df69ba
*color6:	#35a77c
*color14:	#35a77c
*color7:	#5c6a72
*color15:	#5c6a72
URxvt*borderColor:	#f8f0dc
URxvt.depth:	24
URxvt*lineSpace:	0
URxvt*letterSpace:	0
URxvt.iso14755:	false
URxvt.iso14755_52:	false
URxvt*geometry:	209x49
URxvt.scrollBar:	false
URxvt*loginShell:	true
URxvt.internalBorder:	44
URxvt.skipBuiltinGlyphs:	false
URxvt.perl-ext:	default,matcher,selection-to-clipboard,tabbedex,searchable-scrollback,newterm,config-reload,vim-scrollback
URxvt.url-launcher:	linkhandler
URxvt.matcher.button:	1
URxvt.keysym.M-r:	perl:matcher:last
URxvt.keysym.S-C-U:	perl:matcher:list
URxvt.keysym.M-l:	perl:matcher:select
URxvt.saveLine:	10000
URxvt.cursorBlink:	true
URxvt.cursorUnderline:	true
URxvt*inheritPixmap:	false
URxvt*transparent:	false
URxvt.termName:	rxvt-unicode-256color
URxvt*scrollTtyOutput:	false
URxvt*scrollWithBuffer:	true
URxvt*scrollTtyKeypress:	true
URxvt.keysym.Control-l:	command:\033c
URxvt.keysym.Control-Shift-T:	perl:newterm
URxvt.keysym.Shift-Up:	command:\033]720;1\007
URxvt.keysym.Shift-Down:	command:\033]721;1\007
URxvt.secondaryScreen:	1
URxvt.secondaryScroll:	0
URxvt.secondaryWheel:	1
URxvt.keysym.Control-Meta-c:	builtin-string:
URxvt.keysym.Control-Meta-v:	builtin-string:
URxvt.keysym.Shift-Control-V:	eval:paste_clipboard
URxvt.keysym.Shift-Control-C:	eval:selection_to_clipboard
URxvt.tabbedex.autohide:	true
URxvt.tabbedex.tabbar-fg:	2
URxvt.tabbedex.tabbar-bg:	0
URxvt.tabbedex.tab-fg:	3
URxvt.tabbedex.tab-bg:	0
URxvt.tabbedex.no-tabbedex-keys:	true
URxvt.keysym.Control-t:	perl:tabbedex:new_tab
URxvt.keysym.Control-Tab:	perl:tabbedex:next_tab
URxvt.keysym.Control-Shift-Tab:	perl:tabbedex:prev_tab
URxvt.keysym.Control-Shift-Left:	perl:tabbedex:move_tab_left
URxvt.keysym.Control-Shift-Right:	perl:tabbedex:move_tab_right
URxvt.keysym.Control-Shift-R:	perl:tabbedex:rename_tab
urxvt.vim-scrollback:	M-s
urxvt.vim-scrollback-paste:	M-p
urxvt.vim-scrollback-bg:	2
urxvt.vim-scrollback-fg:	10
xterm.vt100.termName:	xterm-256color
xterm.vt100.scaleHeight:	1.0
XTerm.vt100.geometry:	80x32
xterm.vt100.internalBorder:	44
xterm.vt100.borderWidth:	0
XTerm*loginshell:	true
XTerm*savelines:	16384
XTerm*charClass:	33:48,36-47:48,58-59:48,61:48,63-64:48,95:48,126:48
XTerm.vt100.metaSendsEscape:	true
xterm.vt100.utf8:	true
XTerm.vt100.locale:	false
xterm.vt100.scrollBar:	false
xterm.vt100.cursorBlink:	true
xterm.vt100.selectToClipboard:	true
xterm.vt100.trimSelection:	true
xterm.vt100.renderFont:	true
xterm.vt100.wideChars:	true
xterm.vt100.visualBell:	false
xterm.vt100.forceBoxChars:	true
xterm.vt100.font-linedrawing:	true
XTerm.vt100.backarrowKey:	false
XTerm.ttyModes:	erase ^?
*VT100*translations:	#override Shift <Btn1Up>: exec-formatted("linkhandler '%t'", PRIMARY)
xterm.vt100.translations:	#override \n Shift Ctrl <KeyPress> v:insert-selection(CLIPBOARD) \n Shift Ctrl <KeyPress> c:select-set(CLIPBOARD)
XTerm*limitFontsets:	50
XTerm*faceSize:	9
XTerm*faceName:	xft:berry
xterm.vt100.faceNameDoublesize:	Kochi Gothic, Symbola
xterm.vt100.assumeAllChars:	true
XTerm*cjkWidth:	false
XTerm*scrollTtyOutput:	false
st.tabspaces:	4
st.borderpx:	44
```

I've noticed that if I run `sudo make clean install` before subsequent locks on the current session, the message appears immediately. This is turning out to be a really strange bug

**Comment by Barbarossa93** [comment]
Er, I have to run `sudo make uninstall` and then `sudo make clean install` before the message appears immediately

**Comment by Barbarossa93** [comment]
Wait, neverrmind. Apparently it doesn't matter. It seems almost totally random? What on earth is happening here?

**Comment by bakkeby** [comment]
That is bizarre.

**Comment by Barbarossa93** [comment]
Just out of curiousity, does the message show on the PAM screen for you?

**Comment by bakkeby** [comment]
It does yes.

**Comment by Barbarossa93** [comment]
That's interesting, because I don't think it shows the message on the PAM screen even if it appears on the INIT screen.

Since the original issue was solved, I'll close this and open a new one for this new issue
